clcache.py

```diff
@@ -73,6 +73,13 @@ def filesBeneath(path):
             yield os.path.join(path, filename)
 
 
+def subdirectories(path, absolute=True):
```

Comment:
I think "direct" and "child" are redundant (a child is a direct descendant per definition, otherwise it would be a "grand child", "grand grand child" etc.). However, "childDirectories" seemed like a very uncommon name to me.

My personal style is to use "files" and "subdirectories" as the non-recursive ones, and add a 'Beneath' for the functions which get all descendants, not just children.